src/plugins/spec/spec.c

```diff
@@ -578,7 +589,298 @@ static int handleErrors (Key * parentKey, KeySet * ks, Key * key, Key * specKey,
 	return ret;
 }
 
-static void copyMeta (Key * key, Key * specKey)
+static void matchedKeyCopyMeta (Key * key, Key * specKey, Key * parentKey, const char * name)
```

Comment:
Docu is still missing and this function is way too long. Please split it or at least comment the most important pieces.